### PR TITLE
Add permalinks example

### PR DESCRIPTION
**Pull request by julianrubisch**
### Update

this PR now depends on https://github.com/hopsoft/stimulus_reflex/pull/175

### CODE REVIEW

File: app/javascript/controllers/query_string_controller.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ */
+export default class extends ApplicationController {
+  filter () {
+    this.stimulate('QueryStringReflex#filter')
+  }
+
+  afterReflex (element, reflex, error) {
+    if (!error) {
+      const params = new URLSearchParams(window.location.search.slice(1))
+      params.set('category', element.value)
```
Comment (leastbad) [inline app/javascript/controllers/query_string_controller.js]: `Object.keys(Object.assign({}, this.element.dataset)).filter(attr => attr.startsWith(this.identifier)).forEach(attr => { this.element.dataset[attr] <- do cool stuff here })`

Comment (julianrubisch) [reply app/javascript/controllers/query_string_controller.js]: I thnk we're slowly getting there...

**Comment by julianrubisch** [comment]
This one is ready for review. Took a little wiggling around, and I wanted to also show off how button groups can be used. Enjoy!

**Comment by hopsoft** [comment]
Thanks for the addition. I'll test and review this today.

**Comment by julianrubisch** [comment]
done

**Comment by julianrubisch** [comment]
rebased to master

**Comment by julianrubisch** [comment]
merged master into this

**Pull request merged**